racket/collects/racket/private/pre-base.rkt

```diff
@@ -121,26 +109,54 @@
                (random x)]
               [(x y)
                ;; two args, either max and prng, or min and max
-               (cond [(exact-positive-integer? y) ; min and max case
-                      (enforce-random-int-range x)
-                      (enforce-random-int-range y)
-                      (enforce-greater x y)
-                      (+ x (random (- y x)))]
+               (cond [(exact-integer? y) ; min and max case
+                      (unless (exact-integer? x)
+                        (raise-argument-error 'random "exact-integer?" 0 x y))
+                      (unless (< x y)
+                        (raise-argument-error
+                         'random
+                         (string-append "(>/c " (number->string x) ")")
+                         1 x y))
+                      (let ([d (- y x)])
+                        (unless (<= d 4294967087)
+                          (raise-arguments-error
+                           'random
+                           "difference between arguments is greater than 4294967087"
+                           "min" x "max" y))
+                        (+ x (random d)))]
                      [(pseudo-random-generator? y) ; int and prng case
-                      (enforce-random-int-range x)
+                      (unless (and (exact-integer? x)
+                                   (<= 1 x 4294967087))
+                        (raise-argument-error
+                         'random "(integer-in 1 4294967087)" 0 x y))
                       (random x y)]
                      [else
+                      (unless (exact-integer? x)
```

Comment:
This check is new, and maybe unnecessary, and incidentally the only duplicate call to `raise-...-error` (all the others have unique error messages or arguments). 

But it's here because it makes sense to me that `(random "a" "b")` blames its first argument.